lib/rules/filename-is-valid-accessible-name-matches.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Matches the below:
+ * - `HTMLInputElement` with `type=image`
+ * - Element with `role=img`
+ * - is not `aria-hidden`
+ * - has `accessible name`
+ */
+
+const { utils, commons } = axe;
+const { aria, text } = commons;
+
+const isHidden = utils.isHidden(node);
+if (isHidden) {
+	return false;
+}
+
+const accText = text.accessibleText(node);
+if (!accText) {
+	return false;
+}
+
+const srcValue = node.getAttribute('src');
+if (!srcValue) {
+	return false;
+}
+
+const nodeName = node.nodeName.toUpperCase();
+const nodeType = node.getAttribute('type');
```

Comment:
Should use the property lookup `node.type` so it uses the browsers interpreted type (then you don't have to check for it's existence either)